pyanaconda/modules/storage/bootloader/grub2.py

```diff
@@ -363,14 +363,23 @@ def write_config(self):
             if rc:
                 log.error("failed to set menu_auto_hide=1")
 
-        # now tell grub2 to generate the main configuration file
-        rc = util.execWithRedirect(
-            "grub2-mkconfig",
-            ["-o", self.config_file],
-            root=conf.target.system_root
-        )
+        # Executes the grub2-common posttrans script to generate grub config files
+        # This should be autohandled by grub2-common, but posttrans scriptlets
+        # in live image scenarios are not triggered.
+        # FIXME: https://bugzilla.redhat.com/show_bug.cgi?id=2327644
+
+        # Drop the first line as it's the posttrans scriptlet header
+        script = util.execWithCapture("rpm", ["-q", "--scripts", "grub2-common"], root=conf.target.system_root).splitlines()[1:]
+        script = "\n".join(script)
+
+        # Write the script to a temporary file in the chroot
+        with open(join_paths(conf.target.system_root, "/tmp/grub2-posttrans.sh"), "w") as f:
+            f.write(script)
+
+        # Run the script in the chroot
+        rc = util.execWithRedirect("/bin/sh", ["/tmp/grub2-posttrans.sh"], root=conf.target.system_root)
```

Comment:
I think the main issue with this would be that is basically a black box & implementation detail of the GRUB 2 package - eq. some changes unless properly coordinated with us could potentially break things due to changes in GRUB, its packaging or Anaconda. Also unlike a more widely used tool, it would be IMHO quite hard (to impossible) to test, especially outside of this very specific environment.

In comparison, the proposed "shared script" solution could be called by this scriplet (making it much simpler), could be versions (with both GRUB 2 & Anaconda being able to see & control the installed version) as well as presumably being much easier to test a debug separately ("this folder is now /boot, do your thing").